lib/BackgroundJob/Tasks/ImageProcessingTask.php

```diff
@@ -383,4 +384,30 @@ private function calculateMaxImageArea(): int {
 		$maxImageArea = intval((0.75 * $allowedMemory) / 1024); // in pixels^2
 		return $maxImageArea;
 	}
+
+	/**
+	 * Get a path to either the local file or temporary file
+	 *
+	 * @param File $file
+	 * @param int $maxSize maximum size for temporary files
+	 * @return string
+	 */
```

Comment:
This function is taked from IPreview\ProviderV2 that is new but all internal functions exists since OC 9

https://github.com/nextcloud/server/blob/12bdb95c1a40170652570decef7f466e4b97aa65/lib/private/Preview/ProviderV2.php#L81

As it comment, it returns the local file path, or a temporary file path downloaded from the original instance. :grimacing: 

..and this is all the magic .. You can read the files of any storage. :smile: